api/client.go

```diff
@@ -337,10 +338,15 @@ func (c *Config) configureTLS(t *TLSConfig) error {
 	if t.TLSServerName != "" {
 		clientTLSConfig.ServerName = t.TLSServerName
 	}
+	c.clientTLSConfig = clientTLSConfig
 
 	return nil
 }
 
+func (c *Config) TLSConfig() *tls.Config {
+	return c.clientTLSConfig.Clone()
```

Comment:
Does this method need

```go
c.modifyLock.RLock()
defer c.modifyLock.RUnlock()
```

at the beginning, since `c.clientTLSConfig` is being referenced?

Reply:
Wouldn't hurt!